$OpenBSD: patch-src_pcscdaemon_c,v 1.1 2015/10/20 09:57:23 dcoppa Exp $

commit 79410aac02a370d6d92106c015f01ec1491d31ac
Author: Ludovic Rousseau <ludovic.rousseau@free.fr>
Date:   Sun Aug 9 10:21:49 2015 +0200

pcscd: do not write to a closed file handle

In case of receiving a signal to terminate after init has been
completed, at_exit() is called and tries to write to a closed pipe.

This patch simply set pipe to -1 in order to invalidate it after
closing. Pipe value checks are already present in the code.

--- src/pcscdaemon.c.orig	Tue Oct 20 10:39:13 2015
+++ src/pcscdaemon.c	Tue Oct 20 10:39:34 2015
@@ -681,6 +681,7 @@ int main(int argc, char **argv)
 			Log2(PCSC_LOG_ERROR, "write() failed: %s", strerror(errno));
 		}
 		close(pipefd[1]);
+		pipefd[1] = -1;
 	}
 
 	SVCServiceRunLoop();
